add variables to docker compose

firstdorsal opened the pull request:
add variables to docker compose
see: #1376

Commit:
add variables to compose

Commit:
32 char notify

Commit:
fix ports

kiwicopple commented on docker/docker-compose.yml:
```diff
@@ -1,81 +1,88 @@
+    container_name: supabase-auth
+    image: supabase/gotrue:latest
+    ports:
+      - $AUTH_PORT
```
looks like these need brackets right?:

https://docs.docker.com/compose/environment-variables/#substitute-environment-variables-in-compose-files

eg: `${AUTH_PORT}`

firstdorsal replied:
both is possible. i couldn't find documentation on what would be the advantage of using or not using brackets, besides my personal preference of not using them becaus of simplicity/looks reasons.

kiwicopple replied:
> both options are possible

I couldn't find any docs showing it without brackets - I'm just worried it's not supported by some platforms/tools?

Otherwise this PR is great @firstdorsal. If we can change the vars to `${}` (or you can point me to some docs without brackets) then I'm excited to merge this - huge improvement for DX

firstdorsal replied:
no problem :) i added the brackets.
i'm also working on a traefik compose.

kiwicopple replied:
OK this looks great now 🚀 

> I'm also working on a traefik compose.

I noticed the new file in the PR. WIll you also add docs for this one? I'm not familiar with traefik, so I won't be able to document the changes myself

firstdorsal replied:
i provided the readme file for the directory.
i also can provide further documentation where needed.

Commit:
add brackets and kong port

Commit:
sort files and add traefik compose file

firstdorsal commented:
8d88b52 fixes the naming of the supabase docker-compose project to not be named "docker" but supabase.
it's also adding a super easy to deploy traefik example.

Commit:
add pgadmin to traefik compose

firstdorsal commented:
@kiwicopple 
1. the pgadmin folder contains the configuration for the pgadmin container that i added to the traefik compose. since there is no supabase interface provided for the self hosted variant *yet* i added this container to have at least some interface for it.
2. sure i will provide a simple getting started guide. can this then be provided in the docs? https://supabase.io/docs/guides/self-hosting

kiwicopple commented:
> pgadmin container that i added to the traefik compose

Is it possible to move this into the `traefic` folder? If I am asking "what's this folder" then I imagine everyone else will be too. You're right that we're still working on the UI - but we also don't want to "force" people to use  pgadmin if they prefer something like dbeaver, tableplus etc

> sure i will provide a simple getting started guide. can this then be provided in the docs?

Could you please document it into the [traefic Readme](https://github.com/supabase/supabase/blob/master/docker/supabase-traefik/README.md)? If it's on the main docs it will need to be "officially supported", but we don't have anyone yet who can do this (and the docker setup still has manay iterations with Storage+workflows).
